### PR TITLE
fix: validate space ID (bis)

### PR DESCRIPTION
**Pull request by wa0x6e**
Reopening same PR as #417 

But with an additional fix to validate space ID only for offchain spaces

### Test

- See #417 
- With this PR fix, we check space ID only if it's an ID sent by v1 (without network prefix), or if it's an offchain ID sent by v2 (with `s:` prefix)

### CODE REVIEW

File: src/ingestor.ts [hunk 1]
```diff
@@ -79,6 +80,16 @@ export default async function ingestor(req) {
     if (!Object.keys(hashTypes).includes(hash)) return Promise.reject('wrong types');
     type = hashTypes[hash];
 
+    try {
+      if (
+        (message.space.startsWith('s:') || !message.space.includes(':')) &&
+        ensNormalize(message.space) !== message.space.toLowerCase()
```
Comment (ChaituVR) [inline src/ingestor.ts:86]: I think v2 should send requests with just space id without the `s:` prefix

File: src/ingestor.ts [hunk 2]
```diff
@@ -79,6 +80,16 @@ export default async function ingestor(req) {
     if (!Object.keys(hashTypes).includes(hash)) return Promise.reject('wrong types');
     type = hashTypes[hash];
 
+    try {
+      if (
+        (message.space.startsWith('s:') || !message.space.includes(':')) &&
```
Comment (ChaituVR) [inline src/ingestor.ts:85]: Not sure how we missed before but it should throw an error if `message.space` is undefined no?

File: src/ingestor.ts (hunk 1 above)
Comment (ChaituVR) [inline src/ingestor.ts:86]: Maybe we can move to `getSpace`?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/ingestor.ts (hunk 1 above)
Comment (wa0x6e) [reply src/ingestor.ts:86]: We can't, as it's required to differentiate different networks.

Sequencer also accepts space id like `sn:0x0000` and `sep:0x000`

File: src/ingestor.ts (hunk 2 above)
Comment (wa0x6e) [reply src/ingestor.ts:85]: I believe the schema validation is already ensuring `message.space` is always set

File: src/ingestor.ts (hunk 1 above)
Comment (wa0x6e) [reply src/ingestor.ts:86]: See https://github.com/snapshot-labs/snapshot-sequencer/pull/417#issuecomment-2255396997

File: src/ingestor.ts (hunk 2 above)
Comment (ChaituVR) [reply src/ingestor.ts:85]: No 🙈 it is optional. but something else is preventing it not to throw an error 🤔

File: src/ingestor.ts (hunk 1 above)
Comment (ChaituVR) [reply src/ingestor.ts:86]: lolol

File: src/ingestor.ts (hunk 2 above)
Comment (ChaituVR) [reply src/ingestor.ts:85]: I will run few tests and see. but else it looks good, we can merge

Comment (ChaituVR) [reply src/ingestor.ts:85]: I tried to edit profile with this change (alias request) and it fail 
```ts
undefined
TypeError: Cannot read properties of undefined (reading 'startsWith')
```

Comment (ChaituVR) [reply src/ingestor.ts:85]: how it was working before 😅 with https://github.com/snapshot-labs/snapshot-sequencer/pull/417/files

Comment (wa0x6e) [reply src/ingestor.ts:85]: Maybe we could skip the check if `!message.space`

**Review by ChaituVR — APPROVED** [review]
tAck

**Pull request merged**